contentpal/src/main/java/org/dmfs/android/contentpal/operations/Empty.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.operations;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.Operation;
+import org.dmfs.android.contentpal.SoftRowReference;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * An {@link Operation} which fails if the operation affects (updates, deletes or asserts) any rows at all.
+ *
+ * @author Marten Gajda
+ */
+public final class Empty<T> implements Operation<T>
```

Comment:
Is the idea for this class to be used along with `Assert` for rows that should not be in the db?
Could we maybe have dedicated `Assert` classes for that, like `AssertOneRow` which would be composed of an `Assert` and `Counted(1)` and something like `AssertNoRow` which would be `Assert` and `Counted(0)`?
So I am just wondering if this is `Empty` is needed to stand on its own if this is only going to be used along with `Assert`, also because the name is not obvious for me. Or does it refer to empty table? But then again, we could have a `AssertEmpty(Table)` class for that combining `Assert` without data, and `Count(0)`.

Reply:
Right, I guess the only use case for this is asserts. I'll remove it.